src/concurrent.rs

```diff
@@ -289,33 +300,58 @@ mod tests {
         assert_eq!(0, reader2.read(&mut buffer).expect("Expected 0 byte count as EOF after close"));
     }
 
-    #[test]
-    fn test_shareable_file_close_unblocks_reads_without_error() {
+    fn try_shareable_file_close_unblocks_reads_without_error() {
         let dir = tempfile::tempdir().unwrap();
         let path = dir.path().join("pipe");
         unistd::mkfifo(&path, sys::stat::Mode::S_IRUSR | sys::stat::Mode::S_IWUSR).unwrap();
 
-        // We need to write to the FIFO for the read-only open below to not block.
+        // We need to open the FIFO for writes so that the open for reads doesn't block.  But we
+        // have to do this open in a separate thread because it will block too.
+        let (writer_unblock_tx, writer_unblock_rx) = mpsc::channel();
         let writer_handle = {
             let path = path.clone();
             thread::spawn(move || {
-                let mut writer = fs::File::create(&path).unwrap();
-                write!(writer, "some text longer than we'll ever read")
+                let _keep_open = fs::File::create(path).unwrap();
```

Comment:
What I find more clear then relying on this living for the scope is to put an explicit [`std::mem::drop(keep_open)`](https://doc.rust-lang.org/std/mem/fn.drop.html) after the `recv` call.

Reply:
Interesting. Redid this a bit though to simplify what the thread does, but still applied your suggestion.